### PR TITLE
test results in monorepo

### PR DESCRIPTION
**Pull request by nmalyschkin**
I've previously merged the test results widget code as is, so this PR is only the adaptive changes needed to get it working in the monorepo.

### CODE REVIEW

File: packages/esm-patient-test-results-app/src/withDashboardRouting.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { Switch, Route, BrowserRouter, useParams } from "react-router-dom";
+
+export default function withDashboardRouting(WrappedComponent) {
+  const WrappedWithParams = (props) => {
+    const params = useParams();
+    return <WrappedComponent {...props} {...params} />;
+  };
+
+  return ({ basePath, patient, patientUuid }) => (
+    <BrowserRouter basename={`${window.spaBase}${basePath}/test-results`}>
```
Comment (FlorianRappl) [inline packages/esm-patient-test-results-app/src/withDashboardRouting.tsx:11]: I think this whole component should change.

The `wrappedWithParams` is imho no longer required. Using the general navigation context also is now deprecated.

File: packages/esm-patient-test-results-app/src/overview/recent-overview.component.tsx [hunk 2]
```diff
@@ -2,10 +2,12 @@ import React from "react";
 
 import Button from "carbon-components-react/lib/components/Button";
 import DataTableSkeleton from "carbon-components-react/lib/components/DataTableSkeleton";
+
+import { switchTo, useCurrentPatient } from "@openmrs/esm-framework";
```
Comment (FlorianRappl) [inline packages/esm-patient-test-results-app/src/overview/recent-overview.component.tsx:6]: Would not use `useCurrentPatient`. The props already supply the current patient.

File: packages/esm-patient-test-results-app/src/withDashboardRouting.tsx (hunk 1 above)
Comment (nmalyschkin) [reply packages/esm-patient-test-results-app/src/withDashboardRouting.tsx:11]: okay, so what would be the proper alternative to extract parameter?

Comment (FlorianRappl) [reply packages/esm-patient-test-results-app/src/withDashboardRouting.tsx]: The props are passed in. There is no need to extract parameters.

Comment (nmalyschkin) [reply packages/esm-patient-test-results-app/src/withDashboardRouting.tsx]: Do you mean, that the complete is passed through as a prop and I can extract the parameters inside my component?
In need the uuids encoded in the URL in any case or the routing would not work.

Comment (FlorianRappl) [reply packages/esm-patient-test-results-app/src/withDashboardRouting.tsx]: There is no URL anymore. You open the workspace by attaching an extension.

Comment (nmalyschkin) [reply packages/esm-patient-test-results-app/src/withDashboardRouting.tsx]: This is not a workspace anymore but a dashboard (hence the name `withDashboardRouting`).
I used [esm-patient-notes-app](https://github.com/openmrs/openmrs-esm-patient-chart/blob/feature/monorepo/packages/esm-patient-notes-app/src/notes/notes.component.tsx) as reference.

Comment (FlorianRappl) [reply packages/esm-patient-test-results-app/src/withDashboardRouting.tsx]: Then this was even the situation beforehand; props such as `patientUuid` as passed in directly. No need to do anything.

Comment (FlorianRappl) [reply packages/esm-patient-test-results-app/src/withDashboardRouting.tsx]: The reason for the notes provider was that there are components within components that require complex objects such as the patient, which are not transported via the URL. While prop drilling could be done these more complex objects could also be transported via a context.

**Pull request merged**